### ISSUE TITLE
Can't upgrade to docusaurus 3.5.0

### ISSUE THREAD

**Issue by lpillonel** [typesense/docusaurus-theme-search-typesense#49]
## Description

`docusaurus-theme-search-typesense` does not work with `docusaurus` latest `3.5.0` version just released.

## Steps to reproduce

- create a new project `npm init`
- install docusaurus `npm i @docusaurus/core`
- install docusaurus-theme-search-typesense `npm i docusaurus-theme-search-typesense`

## Expected Behavior

As last docusaurus update is not a "breaking change" one, this plugin should be working correctly with it.

## Actual Behavior

Can't install the plugin

```bash
npm i docusaurus-theme-search-typesense
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: caca@1.0.0
npm error Found: @docusaurus/core@3.5.2
npm error node_modules/@docusaurus/core
npm error   @docusaurus/core@"^3.5.2" from the root project
npm error
npm error Could not resolve dependency:
npm error peer @docusaurus/core@"3.4.0" from docusaurus-theme-search-typesense@0.20.0
npm error node_modules/docusaurus-theme-search-typesense
npm error   docusaurus-theme-search-typesense@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error ~/.npm/_logs/2024-08-14T13_55_49_205Z-eresolve-report.txt

npm error A complete log of this run can be found in: ~/.npm/_logs/2024-08-14T13_55_49_205Z-debug-0.log
```

**Comment by joostdecock** [comment]
I have the same problem, but docusaurus has moved on to 3.5.2 now.

If it needs to be locked to the exact docusaurus version -- not uncommon for a plugin -- would it perhaps make sense to adopt their version.
This way if you are running docusaurus `x.y.z` you know you'll be ok if you also run docusaurus-theme-search-typesense `x.y.z`.

**Comment by Kylinus** [comment]
When will the new version be released?

**Comment by jasonbosco** [comment]
Just released this in v0.22.0-1.

**Comment by lpillonel** [comment]
Thanks @jasonbosco !
